tests/_test_msui/test_msui.py

```diff
@@ -328,3 +333,101 @@ def test_flight_track_io(self, mockload, mocksave, mockq, mocki, mockw):
         assert self.window.listFlightTracks.count() == 2
         assert os.path.exists(self.save_ftml)
         os.remove(self.save_ftml)
+
+
+class Test_MSUIMainWindow:
+    def test_storing(self, qtbot):
+        """Test the full scenario: create flight track, open TopView, modify settings,
+           save on close, and restore settings on reopen."""
+        window = msui_mw.MSUIMainWindow()
+        window.show()
+        qtbot.wait_exposed
+
+        window.create_new_flight_track()
+        assert window.listFlightTracks.count() == 1
+        flight_track = window.active_flight_track
+        assert "new flight track (1)" in window.activated_flight_tracks
+
+        waypoint_model = flight_track
+        waypoint_model.insertRows(0, 2, waypoints=[
+            ft.Waypoint(lat=34.44, lon=56.67, location="point1"),
+            ft.Waypoint(lat=77.77, lon=98.67, location="point2")
+        ])
+
+        window.create_view("topview", flight_track)
+        assert window.listViews.count() == 1
+        top_view1 = window.listViews.item(0).window
+        assert isinstance(top_view1, MSUITopViewWindow)
+        assert top_view1.view_type == "Top View"
+        top_view1.cbChangeMapSection.setCurrentText("00 global (cyl)")
+        wms_settings1 = {
+            "url": "http://open-mss.org/",
+            "layer": "ecmwf_EUR_LL015.PLRelHum01",
+            "level": "200.0",
+            "styles": "",
+            "init_time": "2012-10-17T12:00:00Z",
+            "valid_time": "2012-10-17T12:00:00Z"
+        }
+        qtbot.wait(500)
```

Comment:
In general time is not a synchronization primitive and using it in this way is a race condition. That's why we don't use qtbot.wait anywhere except for one location where we haven't come up with a better fix yet. It is better to explicitly wait for the required event to happen and then move on, e.g. with qtbot.wait_until. What exactly is this waiting for?